### DISCUSSION TITLE
Do not impose a limit to the number of heroes that flee or surrender and that can be recruited

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#3199] [Ideas]
In the original game, only two heroes who have fled or surrendered can be recruited.

Why limit the player with this action.
It can be frustrating, especially if the player pays a large amount of gold to get his heroes and armies back and he can't recruit them.

I suggest that there should be "no limit" to the number of heroes that flee or surrender and that can be recruited.

This can possibly be an option.

Related issue: https://github.com/ihhub/fheroes2/issues/3191.

**Comment by Branikolog** [comment]
Hi! 
I expect this only as optional feature. 
As on my point of view, having a limitation of heroes being fled is a part of original heroes gameplay. We have to plan how many heroes we can place in tavern... And how many other heroes sacrifice if we have not enough gold to recruit them back...

**Reply by oleg-derevenetz** [reply]
Hi,

> As on my point of view, having a limitation of heroes being fled is a part of original heroes gameplay.

Agreed. I watch Twitch streams on H3 from time to time and pro gamers are very careful about that (but sometimes they fail anyway) :)

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

I was sure you would want this as an option. :D

**Reply by LeHerosInconnu** [reply]
Remember that we will have a killer AI. :D

**Reply by Branikolog** [reply]
Hi, @LeHerosInconnu 
"Killer AI" will be a part of highest difficulties. So the higher difficulty was set the more experienced player should be. ;) 
All HoMM players are used to the current "flee mechanics". So the feature you've proposed will make their life easier... I'm not sure experienced player would like to make his game easier... (It's just my point of view, anyway.)
There could be much more useful features to implement... Adding such minor features will completely "wogify" this project, which is unwanted... (Considering its current course)
But anyway, in future I expect some kind of such "itemized" release could appear.

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

Some new Heroes 2 players might want some challenge while playing the game.
And certain mechanisms could surprise and frustrate them.
Sometimes, and even for experienced players, it is also difficult to predict events.
Let's say that three of your heroes that you want to keep are attacked during the AI's turn.
For the three combats, you pay a sum of gold to be able to recruit them the next turn when you have the hand (you can't recruit heroes during the AI's turn).
The next turn starts, but you can only recruit the last two of the three heroes you spent gold on, and besides you would have preferred to be able to recruit the first and the third heroes.

I just try to plan for as many eventualities as possible to give all types of players the best possible gaming experience.

**Comment by LeHerosInconnu** [comment]
Here's an example where three heroes were attacked during the AI's turn, and only two of them were available for recruitment at the tavern due to the limitation.
Ambrose, Wilfrey and Ruby had to escape from the combat.
Only Ruby and Wilfrey were available for recruitment at the tavern.
This sometimes happens, and can be a source of frustration for some players, even more so if the player has had to pay a significant amount of gold for them to escape.

<details>
<summary>In fheroes2: (click to expand)</summary>
<br>

![Three heroe tavern 11 001](https://github.com/ihhub/fheroes2/assets/43583994/e0c75771-4a30-4d91-814c-908aa7450770)
![Three heroe tavern 11 002](https://github.com/ihhub/fheroes2/assets/43583994/97768fb5-d15e-404f-93d5-2bd6e5d37a3c)
![Three heroe tavern 11 003](https://github.com/ihhub/fheroes2/assets/43583994/c49a58dc-bc85-4c61-b686-5f157a993df1)
![Three heroe tavern 11 004](https://github.com/ihhub/fheroes2/assets/43583994/f5470092-4bbf-46a2-ad4f-5bc35929fb7e)
![Three heroe tavern 11 005](https://github.com/ihhub/fheroes2/assets/43583994/9ed2411a-c98a-4163-98d5-08201b22f475)
![Three heroe tavern 11 006](https://github.com/ihhub/fheroes2/assets/43583994/191489e5-82d7-458c-bda1-7518cfe2beeb)
![Three heroe tavern 11 007](https://github.com/ihhub/fheroes2/assets/43583994/af08939f-4021-40ed-b486-3ad2fce20880)
![Three heroe tavern 11 008](https://github.com/ihhub/fheroes2/assets/43583994/d3812f31-1a12-4a71-b770-473b765a1332)
![Three heroe tavern 11 009](https://github.com/ihhub/fheroes2/assets/43583994/f26b243d-0a79-4f84-b028-7fd6f5b74ad3)
</details>

Here is the save file:
[Three heroe tavern 11.zip](https://github.com/user-attachments/files/15523477/Three.heroe.tavern.11.zip)

**Comment by LeHerosInconnu** [comment]
This is planned in my mod: https://github.com/ihhub/fheroes2/discussions/10013.
I close the discussion.
